File: ios/Runner/WidgetMethodChannel.swift
import Flutter
import UIKit
import WidgetKit

/// Method channel for widget communication
class WidgetMethodChannel {
    private let channel: FlutterMethodChannel

    init(binaryMessenger: FlutterBinaryMessenger) {
        channel = FlutterMethodChannel(
            name: "com.joonapay.usdc_wallet/widget",
            binaryMessenger: binaryMessenger
        )

        setupMethodHandler()
    }

    private func setupMethodHandler() {
        channel.setMethodCallHandler { [weak self] (call, result) in
            switch call.method {
            case "updateWidget":
                self?.updateWidget()
                result(nil)
            default:
                result(FlutterMethodNotImplemented)
            }
        }
    }

    private func updateWidget() {
        if #available(iOS 14.0, *) {
            WidgetCenter.shared.reloadAllTimelines()
        }
    }
}

File: android/app/src/main/java/com/joonapay/usdc_wallet/WidgetMethodChannel.kt
package com.joonapay.usdc_wallet

import android.content.Context
import com.joonapay.usdc_wallet.widget.BalanceWidgetProvider
import io.flutter.embedding.engine.FlutterEngine
import io.flutter.plugin.common.MethodChannel

/**
 * Method channel for widget communication
 */
class WidgetMethodChannel(
    private val context: Context,
    flutterEngine: FlutterEngine
) {
    private val channel = MethodChannel(
        flutterEngine.dartExecutor.binaryMessenger,
        "com.joonapay.usdc_wallet/widget"
    )

    init {
        setupMethodHandler()
    }

    private fun setupMethodHandler() {
        channel.setMethodCallHandler { call, result ->
            when (call.method) {
                "updateWidget" -> {
                    updateWidget()
                    result.success(null)
                }
                else -> {
                    result.notImplemented()
                }
            }
        }
    }

    private fun updateWidget() {
        BalanceWidgetProvider.updateWidgets(context)
    }
}
